src/Microsoft.AspNetCore.Razor.VSCode/src/Diagnostics/ReportIssueCreator.ts

```diff
@@ -0,0 +1,244 @@
+<details><summary>Expand</summary>
+<p>
+
+\`\`\`C#
+${csharpContent}
+\`\`\`
+
+</p>
+</details>
+
+#### Projected Html document:
+<details><summary>Expand</summary>
+<p>
+
+\`\`\`Html
+${htmlContent}
+\`\`\`
+
+</p>
+</details>
+
+## Machine information
+
+
+**VSCode version**: ${this.vscodeApi.version}
+**Razor.VSCode version**: ${razorExtensionVersion}
+#### \`dotnet --info\`
+
+<details><summary>Expand</summary>
+<p>
+
+\`\`\`
+${dotnetInfo}
+\`\`\`
+
+</p>
+</details>
+
+#### Extensions
+<details><summary>Expand</summary>
+<p>
+
+${extensionTable}
+
+</p>
+</details>`;
+    }
+
+    // Protected for testing
+    protected async getRazor(document: vscode.TextDocument) {
+        const content = document.getText();
+
+        return content;
+    }
+
+    // Protected for testing
+    protected async getProjectedCSharp(razorDocument: IRazorDocument) {
+        let csharpContent = `////////////////////// Projected CSharp as seen by extension ///////////////////////
+${razorDocument.csharpDocument.getContent()}
+
+
+////////////////////// Projected CSharp as seen by VSCode ///////////////////////`;
+
+        const csharpTextDocument = await this.vscodeApi.workspace.openTextDocument(razorDocument.csharpDocument.uri);
+        if (csharpTextDocument) {
+            csharpContent = `${csharpContent}
+${csharpTextDocument.getText()}`;
+        } else {
+            csharpContent = `${csharpContent}
+Unable to resolve VSCode's version of CSharp`;
+        }
+
+        return csharpContent;
+    }
+
+    // Protected for testing
+    protected async getProjectedHtml(razorDocument: IRazorDocument) {
+        let htmlContent = `////////////////////// Projected Html as seen by extension ///////////////////////
+${razorDocument.htmlDocument.getContent()}
+
+
+////////////////////// Projected Html as seen by VSCode ///////////////////////`;
+
+        const htmlTextDocument = await this.vscodeApi.workspace.openTextDocument(razorDocument.htmlDocument.uri);
+        if (htmlTextDocument) {
+            htmlContent = `${htmlContent}
+${htmlTextDocument.getText()}`;
+        } else {
+            htmlContent = `${htmlContent}
+Unable to resolve VSCode's version of Html`;
+        }
+
+        return htmlContent;
+    }
+
+    // Protected for testing
+    protected getExtensionVersion(): string {
+        const extension = this.vscodeApi.extensions.getExtension(razorExtensionId);
+        if (!extension) {
+            return 'Unable to find Razor extension version.';
+        }
+        return extension.packageJSON.version;
+    }
+
+    // Protected for testing
+    protected getInstalledExtensions() {
+        const extensions: Array<vscode.Extension<any>> = this.vscodeApi.extensions.all
+            .filter(extension => extension.packageJSON.isBuiltin === false);
+
+        return extensions.sort(this.sortExtensions);
+    }
+
+    // Protected for testing
+    protected sortExtensions(a: vscode.Extension<any>, b: vscode.Extension<any>): number {
```

Comment:
TBH this feels a bit overcomplicated. Instead of having the `sortExtensions` method and all the unit tests for it, consider replacing L193 with:

```typescript
return extensions.sort((a, b) => 
    a.packageJSON.name.toLowerCase().localeCompare(
        b.packageJSON.name.toLowerCase()
    )
);
```